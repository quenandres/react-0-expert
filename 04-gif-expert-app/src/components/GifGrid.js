import React from 'react'

export const GifGrid = ( { category } ) => {
    
    const getGifs = async () => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=Rick and morty&limit=10&api_key=xJw4xi7Ze2WIz1WMAEkpY9Ppnc7NUkKF';
        const resp = await fetch( url );
        const { data } = await resp.json();
        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images.downsized_medium.url
            }
        })
        console.log(gifs);
    }

    getGifs();

    return (
        <div>
            <h3>{category}</h3>
        </div>
    )
}
