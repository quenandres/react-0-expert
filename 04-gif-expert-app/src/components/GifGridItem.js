import React from 'react'
import '../index.css';

export const GifGridItem = ( {title, url} ) => {

    return (
        <div className="card animate__rubberBand">
            <h1>Test</h1>
            <img src={ url } alt={title} />
            <p> { title } </p>
        </div>
    )
}


