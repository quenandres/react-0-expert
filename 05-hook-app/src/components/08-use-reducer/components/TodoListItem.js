import React from 'react'

export const TodoListItem = ({key, todo, handleToggle, handleDelete}) => {
    return (
        <li 
            className='list-group-item'
            key={ todo.id }
        >
            <p 
                className={`${ todo.done && 'complete' }`}
                onClick={ () => handleToggle(todo.id)}
            > { key + 1 }. { todo.description } </p>
            <button 
                className='btn btn-danger'
                onClick={ () => handleDelete(todo.id) }
            >Borrar</button>
        </li>
    )
}
