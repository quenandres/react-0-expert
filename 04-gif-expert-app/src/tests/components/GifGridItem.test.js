import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

/* 
 1. Enzyme
 2. Enzyme to json
 3. Mostrar el componente correctamente
    * shallow
    * wrapper
    * wrapper .toMatchSnapshot
*/

describe('Pruebas de GifGridItem', () => {
    test('Mostrar correctamente el gifGridItem', () => {
        const wrapper = shallow(<GifGridItem />);
        expect(wrapper).toMatchSnapshot();
    })
    
});