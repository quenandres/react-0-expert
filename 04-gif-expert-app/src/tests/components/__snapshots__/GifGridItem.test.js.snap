// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifGridItem Mostrar correctamente el gifGridItem 1`] = `
<div
  className="card animate__rubberBand"
>
  <h1>
    Test
  </h1>
  <img />
  <p>
     
     
  </p>
</div>
`;
